### PR TITLE
[RA-1 Ch06] 6.3.4 Workload Security

### LINKED ISSUE #1467: [RA-1 Ch06] Workload Security
Develop RA-1 content for Workload Security specified in [RM 7.11.4](https://github.com/cntt-n/CNTT/blob/master/doc/ref_model/chapters/chapter07.md#7114-workload-security)

### PR DESCRIPTION
**Pull request by pgoyal01**
Add content for 6.3.4 Workload Security

Fixes #1467

### CODE REVIEW

**Comment by iangardner22** [comment]
Good work @pgoyal01.  Is it true though with SRIOV / DPDK that Security Groups are not supported?
Do we really need to segregate workloads and storage and network in all cases especially when using tenant networks which by design cannot spill over into other tenants unless you create a shared network or have a common provider?

**Comment by pgoyal01** [comment]
@iangardner22 Do you have pointers to "SRIOV / DPDK that Security Groups are not supported "?

Do we need to segregate networks and storage in all cases? The situation being discussed is the segregation of production and non-production workloads. Given the volatility and experimentation involved with non-production environments, with creation/change/deletion of networks and storage volumes, I think it is prudent to have network and storage segregation.

**Comment by pgoyal01** [comment]
@iangardner22 

"SR-IOV agent only works with NoopFirewallDriver when Security Groups are enabled, but you can still use other firewall_driver for other Agents by updating their conf with the requested firewall driver." [SR-IOV Passthrough for Networking](https://wiki.openstack.org/wiki/SR-IOV-Passthrough-For-Networking).

**Comment by iangardner22** [comment]
> @iangardner22 Do you have pointers to "SRIOV / DPDK that Security Groups are not supported "?
Ah may be wrong on supportability but I was told that Security Groups can impact performance for high performance workloads so are typically disabled... @karinesevilla @samuelhellec is this your experience?

> Do we need to segregate networks and storage in all cases? The situation being discussed is the segregation of production and non-production workloads. Given the volatility and experimentation involved with non-production environments, with creation/change/deletion of networks and storage volumes, I think it is prudent to have network and storage segregation.
OK

**Comment by karinesevilla** [comment]
@iangardner22 @pgoyal01 
About security groups, in our deployments, they are not implemented for SRIOV compute nodes and not recommended for DPDK compute nodes

**Comment by pgoyal01** [comment]
@iangardner22 @karinesevilla Added a subsection [6.3.4.1 SR-IOV and DPDK Considerations](https://github.com/cntt-n/CNTT/blob/pgoyal01-patch-RA1-Workload-Security/doc/ref_arch/openstack/chapters/chapter06.md#6341-sr-iov-and-dpdk-considerations)

Have tired to capture above discussions.

**Comment by karinesevilla** [comment]
@pgoyal01 Agree with the text added in 6.3.4.1
Typo: ussing -> using

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 1]
```diff
@@ -130,6 +129,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
 <a name="6.3.4"></a>
 ### 6.3.4 Workload Security
 
+OpenStack segregates its infrastructure (for example, hosts) by Regions, Host aggregates and Availability Zones (AZ). Workloads can also be segregated by server groups (affinity and non-affinity groups). These options support the workloads placement requirement _sec.wl.001_.
+
+Separation of non-production and production workloads, or by workload category (for example, payment card information, healthcare, etc.) requires separation through server groups (Regions, AZs) but also requires network and storage segregation as in Regions but also AZs if engineered to do so. Thus, the separation of these workloads is handled through placement of workloads in separate AZs and/or Regions (_sec.wl.005_ and _sec.wl.006_).
```
Comment (deebhatia) [inline doc/ref_arch/openstack/chapters/chapter06.md:134]: `...separation through server groups ...` -  Server groups can be misinterpreted as OpenStack affinity and anti-affinity filters which is not what Region and AZs mean. 
Needs a re-wording.

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter06.md:134]: @deebhatia added "for example". Hope that helps. Thanks

Comment (deebhatia) [reply doc/ref_arch/openstack/chapters/chapter06.md]: Maybe it's only me. Let's go ahead with server groups. If someone else raises it again, we may reword it later on.

**Pull request merged**